plugins/audit-log-node/src/DefaultAuditLogger.ts

```diff
@@ -0,0 +1,159 @@
+        'service',
+      )
+        ? credentials.principal.subject
+        : undefined;
+
+      return userEntityRef ?? serviceEntityRef;
+    } catch {
+      return undefined;
+    }
+  }
+  async createAuditLogDetails(
+    options: AuditLogDetailsOptions,
+  ): Promise<AuditLogDetails> {
+    const { eventName, stage, metadata, request, response, status } = options;
+
+    const actorId = options.actorId || (await this.getActorId(request)) || null;
+
+    // Secrets in the body field should be redacted by the user before passing in the request object
+    const auditRequest = request
+      ? {
+          method: request.method,
+          url: request.originalUrl,
+          params: request.params,
+          query: request.query,
+          body: request.body,
+        }
+      : undefined;
+
+    const actor: ActorDetails = { actorId };
+    if (request) {
+      actor.ip = request.ip;
+      actor.hostname = request.hostname;
+      actor.userAgent = request.get('user-agent');
+    }
+
+    const auditLogCommonDetails = {
+      actor,
+      meta: metadata || {},
+      request: auditRequest,
+      isAuditLog: true as const,
+      response,
+      eventName,
+      stage,
+    };
+
+    if (auditRequest) {
+      auditLogCommonDetails.request = auditRequest;
+    }
+    if (response) {
+      auditLogCommonDetails.response = response;
+    }
+
+    if (status === 'failed') {
+      const errs = options.errors as ErrorLike[];
+      return {
+        ...auditLogCommonDetails,
+        status,
+        errors: errs.map(err => {
+          return {
+            name: err.name,
+            message: err.message,
+            stack: err.stack,
+          };
+        }),
+      };
+    }
+
+    return {
+      ...auditLogCommonDetails,
+      status,
+    };
+  }
+  async auditLog(options: AuditLogOptions): Promise<void> {
+    const logLevel = options.level || 'info';
+    const auditLogDetails = await this.createAuditLogDetails({
+      eventName: options.eventName,
+      status: 'succeeded',
+      stage: options.stage,
+      actorId: options.actorId,
+      request: options.request,
+      response: options.response,
+      metadata: options.metadata,
+    });
+    switch (logLevel) {
+      case 'info':
+        this.logger.info(options.message, auditLogDetails as JsonObject);
+        return;
+      case 'debug':
+        this.logger.debug(options.message, auditLogDetails as JsonObject);
+        return;
+      case 'warn':
+        this.logger.warn(options.message, auditLogDetails as JsonObject);
+        return;
+      case 'error':
+        this.logger.error(options.message, auditLogDetails as JsonObject);
+        return;
+      default:
+        throw new Error(`Log level of ${logLevel} is not supported`);
+    }
+  }
+
+  async auditErrorLog(options: AuditErrorLogOptions): Promise<void> {
+    const auditLogDetails = await this.createAuditLogDetails({
+      eventName: options.eventName,
+      status: 'failed',
```

Comment:
Should we hardcode the status in both implementations of the auditLog, or should we require the user to specify whether it was a success or failure? I believe we should make the user specify the outcome for less code duplication.

Reply:
I guess that could make sense to give more control to the user. We can just set the default statuses to `failed` and `succeeded` for the `auditErrorLog` and `auditLog` functions respectively if no status is supplied.

Reply:
Removed the `auditErrorLog` function in https://github.com/janus-idp/backstage-plugins/pull/1730/commits/35eb36afdbce0cb4e8fb5434d00c0e92a1edd7cc in favor of using the `auditLog` function for everything for simplicity. 

This would require the following changes:
- All usages of `auditErrorLog` to be changed to `auditLog`
- A `status` field must be provided to all `auditLog` function calls
- A `level: 'error'` field should be provided to all `auditLog` calls converted from `auditErrorLog`
@AndrienkoAleksandr what is your opinion on this change? Would it severely break your RBAC audit logging?